src/services/item/service.ts

```diff
@@ -388,6 +389,22 @@ export class ItemService {
     // TODO: args?
     const result = await itemRepository.copy(item, actor, parentItem);
 
+    // create a membership if needed
+    await itemMembershipRepository
+      .post({
+        item: result,
+        member: actor,
+        creator: actor,
+        permission: PermissionLevel.Admin,
+      })
+      .catch((e) => {
+        // admin permission already exists and does not need to be added
+        if (e instanceof InvalidMembership) {
+          return;
+        }
+        throw e;
+      });
+
```

Comment:
Shouldn't this be a single DB transaction? (i.e if the second query fails, we don't want ab Item without membership? What about items that were copied before this change?

It's a bit complicated to rebuild all the repositories everywhere when you need a transaction, I don't know if it could be done better

Here we could do something like this:
```
await this.db.transaction(async (manager) => {
      const repositories = buildRepositories(manager)
      const result = await repositories.itemRepository.copy(item, actor, parentItem)
      await repositories.itemMembershipRepository
        .post({
          item: result,
          member: actor,
          creator: actor,
          permission: PermissionLevel.Admin,
        })
        // Ideally here rather catch the specific exception, so that the item can rollback if this is an unexpected exception
        .catch((e) => {
          // admin permission already exists and does not need to be added
        });
    })
```

Reply:
It's already one transaction since it's defined in the `controller`. I felt like creating a membership is "related" to creating an item, that's why I've put it in the item service (it's also because of some habits). But I'm open to put it in the controller. However since we run hooks in copy, if you don't have your membership beforehand some of the hooks (that might check your authorization) will fail.

Reply:
You're right, no need to move.
I actually would expect the controller to only handle general request thing (authentication pre-handler is good example) and HTTP requests details, then only calling one service.
The service can call repositories but also other services, so in my opinion the membership creation is at the right place. 

Eventually I'd think about not creating transaction in controllers but only in services, but that would be for a future change

Reply:
@morganridel I agree, but how do you handle which service should initiate the transaction? E.g. if item service needs the item memberships service, how does it pass the current transaction (also if some other services uses item membership without initiating its own transaction)?